Code/Framework/AzToolsFramework/AzToolsFramework/ViewportUi/ButtonGroup.cpp

```diff
@@ -57,7 +57,15 @@ namespace AzToolsFramework::ViewportUi::Internal
 
     ButtonId ButtonGroup::AddButton(const AZStd::string& icon, const AZStd::string& name)
     {
-        auto buttonId = ButtonId(m_buttons.size() + 1);
+        auto max = ButtonId(0);
```

Comment:
update: using `max` here can lead to problems as some platform have this as a `define` - it might be better to name this `lastButtonId` potentially? Also instead of using a loop, you could use an algorithm 🙂 Have a look at [max_element](https://en.cppreference.com/w/cpp/algorithm/max_element), it should be just what you need

Reply:
I tried using max_element but there is no AZStd version, I tried to make it work with the std version but I couldn't get the predicate to work: 


```
[](AZStd::pair<ButtonId, AZStd::unique_ptr<Button>> buttonPairA, 
AZStd::pair<ButtonId, AZStd::unique_ptr<Button>> buttonPairB)   
{
    return buttonAId.first < buttonBId.first;
};
```
m_buttons is of type `AZStd::unordered_map<ButtonId, AZStd::unique_ptr<Button>> `
Is there anything glaringly wrong?

Reply:
Ah so to add this to AZStd, it's possible to go to algorithm.h (in AzCore/std folder) and add it as many of the other algorithms are exposed with `using max_element;` (might as well add `min_element` and `minmax_element` while you're there too).

With the code, the issue is likely because you're passing a `std::pair` (which contains a `unique_ptr`) by value in the parameter list which will cause a move and won't work. Trying making them both const references, I think that should work then. The other thing is what type is ButtonId (does it provide a `<` operator?)